#**📊 1 - How many total weapons are in the game?** 

**List all weapons found in the dataset, with the understanding that the dataset may not contain every weapon from the game.**

In [0]:
%sql
SELECT weapon_key, weapon_name, category, weight, is_dlc 
FROM eldenringcatalog.gold.dim_weapons 
LIMIT 20;

weapon_key weapon_name category weight is_dlc 0 Dueling Shield Thrusting Shields 9.0 1 1 Carian Thrusting Shield Thrusting Shields 10.5 1 2 Dryleaf Arts Hand-to-Hand Arts 1.0 1 3 Dane's Footwork Hand-to-Hand Arts 1.0 1 4 Smithscript Dagger Throwing Blades 1.5 1 5 Backhand Blade Backhand Blades 2.0 1 6 Smithscript Cirque Backhand Blades 2.0 1 7 Curseblade's Cirque Backhand Blades 4.5 1 8 Firespark Perfume Bottle Perfume Bottles 1.0 1 9 Chilling Perfume Bottle Perfume Bottles 1.0 1 10 Frenzyflame Perfume Bottle Perfume Bottles 1.0 1 11 Lightning Perfume Bottle Perfume Bottles 1.0 1 12 Deadly Poison Perfume Bottle Perfume Bottles 1.0 1 13 Red Bear's Claw Beast Claws 4.0 1 14 Beast Claw (Weapon) Beast Claws 3.0 1 15 Milady Light Greatswords 6.5 1 16 Leda's Sword Light Greatswords 7.5 1 17 Rellana's Twin Blades Light Greatswords 8.0 1 18 Great Katana Great Katanas 9.0 1 19 Dragon-Hunter's Great Katana Great Katanas 10.5 1

In [0]:
%sql
SELECT COUNT(*) AS total_weapon, COUNT(DISTINCT weapon_name) AS unique_weapons FROM eldenringcatalog.gold.dim_weapons

total_weapon unique_weapons 402 402

**Total weapons available: 402**

# **📊 2 - What are all weapon categories available?**

**Identify weapon types to understand combat style diversity.**

In [0]:
%sql
SELECT category, COUNT(*) as weapon_count
FROM eldenringcatalog.gold.dim_weapons
GROUP BY category
ORDER BY weapon_count DESC;

category weapon_count Greatswords 24 Straight Swords 23 Colossal Weapons 20 Curved Swords 19 Spears 19 Glintstone Staffs 18 Daggers 18 Halberds 18 Axes 17 Colossal Swords 16 Great Hammers 16 Hammers 16 Fists 14 Greataxes 14 Sacred Seals 12 Curved Greatswords 11 Great Spears 10 Katanas 10 Crossbows 9 Bows 9 Thrusting Swords 8 Torches 8 Twinblades 8 Whips 7 Heavy Thrusting Swords 6 Flails 6 Claws 5 Reapers 5 Perfume Bottles 5 Greatbows 5 Light Bows 5 Light Greatswords 3 Great Katanas 3 Ballistas 3 Backhand Blades 3 Hand-to-Hand Arts 2 Beast Claws 2 Thrusting Shields 2 Glinstone Staves 2 Throwing Blades 1

In [0]:
%sql
SELECT category, is_dlc, COUNT(*) as count
FROM eldenringcatalog.gold.dim_weapons
GROUP BY category, is_dlc
ORDER BY category, is_dlc;

category is_dlc count Axes 0 13 Axes 1 4 Backhand Blades 1 3 Ballistas 0 2 Ballistas 1 1 Beast Claws 1 2 Bows 0 7 Bows 1 2 Claws 0 4 Claws 1 1 Colossal Swords 0 11 Colossal Swords 1 5 Colossal Weapons 0 15 Colossal Weapons 1 5 Crossbows 0 7 Crossbows 1 2 Curved Greatswords 0 9 Curved Greatswords 1 2 Curved Swords 0 15 Curved Swords 1 4 Daggers 0 16 Daggers 1 2 Fists 0 9 Fists 1 5 Flails 0 5 Flails 1 1 Glinstone Staves 1 2 Glintstone Staffs 0 18 Great Hammers 0 14 Great Hammers 1 2 Great Katanas 1 3 Great Spears 0 6 Great Spears 1 4 Greataxes 0 11 Greataxes 1 3 Greatbows 0 4 Greatbows 1 1 Greatswords 0 21 Greatswords 1 3 Halberds 0 16 Halberds 1 2 Hammers 0 15 Hammers 1 1 Hand-to-Hand Arts 1 2 Heavy Thrusting Swords 0 4 Heavy Thrusting Swords 1 2 Katanas 0 8 Katanas 1 2 Light Bows 0 5 Light Greatswords 1 3 Perfume Bottles 1 5 Reapers 0 4 Reapers 1 1 Sacred Seals 0 9 Sacred Seals 1 3 Spears 0 16 Spears 1 3 Straight Swords 0 19 Straight Swords 1 4 Throwing Blades 1 1 Thrusting Shields 1 2 Thrusting Swords 0 7 Thrusting Swords 1 1 Torches 0 6 Torches 1 2 Twinblades 0 6 Twinblades 1 2 Whips 0 6 Whips 1 1

# **📊  3 - Which weapon category has the most options?**

**Popular categories offer more build variety.**

In [0]:
%sql
SELECT category, COUNT(*) as total_weapons
FROM eldenringcatalog.gold.dim_weapons
GROUP BY category
ORDER BY total_weapons DESC
LIMIT 3;

category total_weapons Greatswords 24 Straight Swords 23 Colossal Weapons 20

In [0]:
%sql
SELECT weapon_name, weight, is_dlc
FROM eldenringcatalog.gold.dim_weapons
WHERE category = 'Greatswords'
LIMIT 10;

weapon_name weight is_dlc Bastard Sword 9.0 0 Claymore 9.0 0 Iron Greatsword 12.0 0 Lordsworn's Greatsword 9.0 0 Knight's Greatsword 10.0 0 Banished Knight's Greatsword 10.0 0 Forked Greatsword 9.0 0 Flamberge 10.0 0 Gargoyle's Greatsword 11.5 0 Gargoyle's Blackblade 11.5 0

# **📊  4- What is the weight distribution of weapons?**

**Weight affects equipment load and mobility.**

In [0]:
%sql
SELECT 
    CASE 
        WHEN weight <= 3 THEN 'Light'
        WHEN weight <= 8 THEN 'Medium'
        WHEN weight <= 15 THEN 'Heavy'
        ELSE 'Very Heavy'
    END AS weight_class,
    COUNT(*) AS weapon_count,
    ROUND(AVG(weight), 2) AS avg_weight
FROM eldenringcatalog.gold.dim_weapons
GROUP BY weight_class
ORDER BY avg_weight;

weight_class weapon_count avg_weight Light 108 2.16 Medium 157 5.35 Heavy 100 10.67 Very Heavy 37 19.5

**Most weapons fall into medium weight range, with extremes for specialized builds.**

# **📊 5 - Which weapons have the lowest stat requirements**

**Early-game viability and accessibility.**

In [0]:
%sql
SELECT 
    weapon_name,
    category,
    is_dlc,
    required_strength,
    required_dexterity,
    required_intelligence,
    required_faith,
    required_arcane,
    (required_strength + required_dexterity + required_intelligence + required_faith + required_arcane) AS total_requirements,
    weight
FROM eldenringcatalog.gold.dim_weapons
ORDER BY total_requirements, weight 
LIMIT 20;

weapon_name category is_dlc required_strength required_dexterity required_intelligence required_faith required_arcane total_requirements weight Serpent-Hunter Great Spears 0 0 0 0 0 0 0 12.0 Frenzied Flame Seal Sacred Seals 0 0 0 0 0 0 0 12.0 Noble's Slender Sword Straight Swords 0 8 0 0 0 0 8 3.5 Lamenting Visage Torches 1 9 0 0 0 0 9 2.0 Torch Torches 0 5 5 0 0 0 10 1.5 Club Hammers 0 10 0 0 0 0 10 3.5 Dagger Daggers 0 5 9 0 0 0 14 1.5 Clawmark Seal Sacred Seals 0 4 0 0 10 0 14 1.5 Finger Seal Sacred Seals 0 4 0 0 10 0 14 1.5 Spiked Spear Spears 0 14 0 0 0 0 14 6.0 Dryleaf Arts Hand-to-Hand Arts 1 8 8 0 0 0 16 1.0 Dane's Footwork Hand-to-Hand Arts 1 8 8 0 0 0 16 1.0 Caestus Fists 0 8 8 0 0 0 16 1.5 Glintstone Staff Glintstone Staffs 0 6 0 10 0 0 16 3.0 Demi-Human Queen's Staff Glintstone Staffs 0 6 0 10 0 0 16 3.0 Firespark Perfume Bottle Perfume Bottles 1 3 14 0 0 0 17 1.0 Celebrant's Sickle Daggers 0 6 11 0 0 0 17 1.5 Weathered Straight Sword Straight Swords 0 7 10 0 0 0 17 3.0 Hand Axe Axes 0 9 8 0 0 0 17 3.5 Grave Scythe Reapers 0 17 0 0 0 0 17 9.5

**Identify beginner-friendly weapons accessible to all starting classes**

# **📊 7 - Which weapons require the HIGHEST total stat investment?**

**Late-game weapons requiring specialized builds.**

In [0]:
%sql
SELECT 
    weapon_name,
    category,
    required_strength,
    required_dexterity,
    required_intelligence,
    required_faith,
    required_arcane,
    (required_strength + required_dexterity + required_intelligence + required_faith + required_arcane) AS total_requirements,
    is_dlc
FROM eldenringcatalog.gold.dim_weapons
ORDER BY total_requirements DESC
LIMIT 20;

weapon_name category required_strength required_dexterity required_intelligence required_faith required_arcane total_requirements is_dlc Anvil Hammer Colossal Weapons 39 10 11 20 0 80 1 Carian Regal Scepter Glintstone Staffs 8 10 60 0 0 78 0 Sword of Night and Flame Straight Swords 12 12 24 24 0 72 0 Greatsword of Radahn (Lord) Colossal Swords 32 24 15 0 0 71 1 Greatsword of Radahn (Light) Colossal Swords 32 24 15 0 0 71 1 Devourer's Scepter Great Hammers 24 20 0 25 0 69 0 Spear of the Impaler Great Spears 14 35 0 18 0 67 1 Ruins Greatsword Colossal Swords 50 0 16 0 0 66 0 Staff of the Avatar Colossal Weapons 34 8 0 24 0 66 0 Fallingstar Beast Jaw Colossal Weapons 34 12 20 0 0 66 0 Maliketh's Black Blade Colossal Swords 34 12 0 20 0 66 0 Morgott's Cursed Sword Curved Greatswords 14 35 0 0 17 66 0 Royal Greatsword Colossal Swords 26 18 22 0 0 66 0 Golden Order Greatsword Greatswords 16 21 0 28 0 65 0 Mohgwyn's Sacred Spear Great Spears 24 14 0 0 27 65 0 Dark Moon Greatsword Greatswords 16 11 38 0 0 65 0 Eclipse Shotel Curved Swords 10 25 0 30 0 65 0 Starscourge Greatsword Colossal Swords 38 12 15 0 0 65 0 Ancient Meteoric Ore Greatsword Colossal Swords 35 10 0 0 19 64 1 Hand of Malenia Katanas 16 48 0 0 0 64 0

**Identify prestigious weapons that require significant stat investment.**

# **7 - How many weapons are DLC vs base game?**

**Understand content distribution across expansions.**

In [0]:
%sql
SELECT 
    CASE WHEN is_dlc = 1 THEN 'DLC' ELSE 'Base Game' END AS content_type,
    COUNT(*) AS weapon_count,
    ROUND(COUNT(*) * 100.0 / (SELECT COUNT(*) FROM eldenringcatalog.gold.dim_weapons), 2) AS percentage
FROM eldenringcatalog.gold.dim_weapons
GROUP BY is_dlc;

content_type weapon_count percentage DLC 94 23.38 Base Game 308 76.62

**See how much weapon variety the DLC adds.**

# **📊 8: What damage types are available?**

**Understanding damage variety for enemy weaknesses.**

In [0]:
%sql
SELECT damage_type, COUNT(*) as weapon_count
FROM eldenringcatalog.gold.dim_weapons
GROUP BY damage_type
ORDER BY weapon_count DESC;

damage_type weapon_count Strike 105 Standard/Pierce 95 Slash/Pierce 49 Standard 48 Slash 41 None 28 Pierce 27 Strike/Pierce 3 LightBows 2 Standar/Pierce 1 - 1 Standart/Pierce 1 StrikeDamage 1

In [0]:
%sql
SELECT weapon_name, category ,is_dlc
FROM eldenringcatalog.gold.dim_weapons
WHERE damage_type = 'Strike'
LIMIT 10;

weapon_name category is_dlc Dryleaf Arts Hand-to-Hand Arts 1 Dane's Footwork Hand-to-Hand Arts 1 Stone-Sheathed Sword Straight Swords 1 Jawbone Axe Axes 0 Club Hammers 0 Curved Club Hammers 0 Spiked Club Hammers 0 Stone Club Hammers 0 Mace Hammers 0 Morning Star Hammers 0

**Variety in damage types (slash, strike, pierce, etc.) for tactical advantages.**

# **📊 9- Which weapons have passive effects?**

**Status effects add strategic value beyond raw damage.**

In [0]:
%sql
SELECT 
    weapon_name,
    category,
    passive_effect,
    weight,
    is_dlc
FROM eldenringcatalog.gold.dim_weapons
WHERE passive_effect != 'No passive effects'
ORDER BY passive_effect, weapon_name;


weapon_name category passive_effect weight is_dlc Bloodfiend's Fork Spears (30) Hemorrhage 4.5 1 Obsidian Lamina Reapers (33) Hemorrhage 6.5 1 Poisoned Hand Fists (36) Poison 1.0 1 Bloodstained Dagger Daggers (38) Hemorrhage 2.0 0 Wakizashi Daggers (38) Hemorrhage 3.0 0 Thiollier's Hidden Needle Fists (42) Sleep 0.5 1 Velvet Sword of St Trina Straight Swords (42) Sleep 2.5 1 Star-Lined Sword Katanas (45) 5.0 1 Staff of the Guilty Glintstone Staffs (45) Boosts Thorn Sorcery Hemorrhage 4.5 0 Beast Claw (Weapon) Beast Claws (45) Hemorrhage 3.0 1 Claws of Night Claws (45) Hemorrhage 3.0 1 Curseblade's Cirque Backhand Blades (45) Hemorrhage 4.5 1 Nagakiba Katanas (45) Hemorrhage 7.0 0 Red Bear's Claw Beast Claws (45) Hemorrhage 4.0 1 Scavenger's Curved Sword Curved Swords (45) Hemorrhage 3.5 0 Spiked Caestus Fists (45) Hemorrhage 2.0 0 Star Fist Fists (45) Hemorrhage 3.0 0 Sword of Night Katanas (45) Hemorrhage 6.5 1 Uchigatana Katanas (45) Hemorrhage 5.5 0 Chainlink Flail Flails (50) Hemorrhage 8.0 0 Cross-Naginata Spears (50) Hemorrhage 8.0 0 Flail Flails (50) Hemorrhage 5.0 0 Forked Hatchet Axes (50) Hemorrhage 2.5 0 Forked-Tongue Hatchet Axes (50) Hemorrhage 4.0 1 Hand of Malenia Katanas (50) Hemorrhage 7.0 0 Inquisitor's Girandole Spears (50) Hemorrhage 7.5 0 Meteoric Ore Blade Katanas (50) Hemorrhage 7.5 0 Moonveil Katanas (50) Hemorrhage 6.5 0 Morning Star Hammers (50) Hemorrhage 5.0 0 Nightrider Flail Flails (50) Hemorrhage 6.0 0 Reduvia Daggers (50) Hemorrhage 2.5 0 Rivers of Blood Katanas (50) Hemorrhage 6.5 0 Spiked Club Hammers (50) Hemorrhage 5.0 0 Spiked Spear Spears (50) Hemorrhage 6.0 0 Thorned Whip Whips (50) Hemorrhage 2.5 0 Rotten Crystal Sword Straight Swords (50) Scarlet Rot 4.5 0 Scorpion's Stinger Daggers (50) Scarlet Rot 2.5 0 Dark Moon Greatsword Greatswords (55) Frostbite 10.0 0 Putrescence Cleaver Greataxes (55) Frostbite 13.5 1 Barbed Staff-Spear Great Spears (55) Hemorrhage 7.5 1 Bloodfiend's Sacred Spear Great Spears (55) Hemorrhage 8.0 1 Bloodhound's Fang Curved Greatswords (55) Hemorrhage 11.5 0 Bloody Helice Heavy Thrusting Swords (55) Hemorrhage 8.0 0 Dragon-Hunter's Great Katana Great Katanas (55) Hemorrhage 10.5 1 Eleonora's Poleblade Twinblades (55) Hemorrhage 6.0 0 Falx Curved Swords (55) Hemorrhage 4.0 1 Flamberge Greatswords (55) Hemorrhage 10.0 0 Forked Greatsword Greatswords (55) Hemorrhage 9.0 0 Grave Scythe Reapers (55) Hemorrhage 9.5 0 Great Katana Great Katanas (55) Hemorrhage 9.0 1 Great Omenkiller Cleaver Greataxes (55) Hemorrhage 11.0 0 Great Stars Great Hammers (55) Hemorrhage 10.0 0 Halo Scythe Reapers (55) Hemorrhage 8.5 0 Hoslow's Petal Whip Whips (55) Hemorrhage 3.5 0 Mohgwyn's Sacred Spear Great Spears (55) Hemorrhage 10.0 0 Rakshasa's Great Katana Great Katanas (55) Hemorrhage 9.5 1 Scythe Reapers (55) Hemorrhage 7.5 0 Sword of Milos Greatswords (55) Hemorrhage 10.0 0 Vulgar Militia Saw Halberds (55) Hemorrhage 8.0 0 Winged Scythe Reapers (55) Hemorrhage 7.5 0 Frenzied Flame Seal Sacred Seals (55) Madness 12.0 0 Madding Hand Fists (55) Madness 1.0 1 Antspur Rapier Thrusting Swords (55) Scarlet Rot 3.0 0 Poleblade of the Bud Halberds (55) Scarlet Rot 10.0 1 Rotten Crystal Spear Spears (55) Scarlet Rot 5.5 0 Chilling Perfume Bottle Perfume Bottles (60) Frostbite 1.0 1 Frozen Needle Thrusting Swords (60) Frostbite 2.5 0 Bloodhound Claws Claws (60) Hemorrhage 3.0 0 Hookclaws Claws (60) Hemorrhage 2.0 0 Morgott's Cursed Sword Curved Greatswords (60) Hemorrhage 7.5 0 Raptor Talons Claws (60) Hemorrhage 1.5 0 Frenzyflame Perfume Bottle Perfume Bottles (60) Madness 1.0 1 Death's Poker Greatswords (65) Frostbite 8.5 0 Ghostflame Torch Torches (65) Frostbite 2.5 0 Icerind Hatchet Axes (65) Frostbite 3.0 0 Zamor Curved Sword Curved Greatswords (65) Frostbite 9.0 0 Nanaya's Torch Torches (65) Madness 3.5 1 Vyke's War Spear Great Spears (65) Madness 8.0 0 Rotten Battle Hammer Great Hammers (65) Scarlet Rot 10.0 0 Rotten Greataxe Colossal Weapons (65) Scarlet Rot 20.0 0 Deadly Poison Perfum

In [0]:
%sql
SELECT passive_effect, COUNT(*) as count
FROM eldenringcatalog.gold.dim_weapons
WHERE passive_effect != 'No passive effects'
GROUP BY passive_effect
ORDER BY count DESC;

passive_effect count (55) Hemorrhage 21 (50) Hemorrhage 16 (45) Hemorrhage 10 (60) Hemorrhage 4 (65) Frostbite 4 (55) Scarlet Rot 3 (66) Poison 3 (65) Madness 2 (55) Frostbite 2 (42) Sleep 2 (38) Hemorrhage 2 (55) Madness 2 (60) Frostbite 2 (65) Scarlet Rot 2 (50) Scarlet Rot 2 (36) Poison 1 (30) Hemorrhage 1 (72) Poison 1 (60) Madness 1 Raises Immunity by 50 1 Boosts Stonedigger Sorcery 1 Boosts Giants' FlameIncantations 1 Boosts Fundamentalist Inc. 1 Boosts Godslayer Incantations 1 (45) Boosts Thorn Sorcery Hemorrhage 1 (75) Scarlet Rot 1 Boosts Death Sorceries 1 Boosts Bestial Incantations 1 : Boosts Lava Sorcery 1 (33) Hemorrhage 1 Boosts full moon sorcery 1 Boosts Crystalian sorceries 1 (70) Hemorrhage 1 : Boosts Carian Sword Sorcery 1 (45) 1 (66) Sleep 1 Rune Gain on Hit 1 Boosts Invisibility Sorcery 1 Boosts Glintblade Sorcery 1 : (38) Hemorrhage 1 Boosts Crystalian Sor. (50) Scarlet Rot 1 Boosts Gravity Sorcery 1 Boosts Dragon Communion Incantations 1 Causes blood loss buildup (65) 1 Roar Buff 1 Boosts Dragon Cult Incantations 1 (72) Sleep 1

**Identify weapons with bleed, frost, poison, etc. for status-based builds.**

# **📊 10 - Load weapon stats fact table and show sample data**

**Access detailed performance metrics for weapons.**

In [0]:
%sql
SELECT 
    weapon_name,
    category,
    upgrade_level,
    attack_physical,
    total_attack_power,
    weight
FROM eldenringcatalog.gold.fact_weapon_stats
LIMIT 20;

weapon_name category upgrade_level attack_physical total_attack_power weight Dueling Shield Thrusting Shields Standard 125.0 125.0 9.0 Dueling Shield Thrusting Shields Standard +1 132.0 132.0 9.0 Dueling Shield Thrusting Shields Standard +2 139.0 139.0 9.0 Dueling Shield Thrusting Shields Standard +3 146.0 146.0 9.0 Dueling Shield Thrusting Shields Standard +4 154.0 154.0 9.0 Dueling Shield Thrusting Shields Standard +5 161.0 161.0 9.0 Dueling Shield Thrusting Shields Standard +6 168.0 168.0 9.0 Dueling Shield Thrusting Shields Standard +7 175.0 175.0 9.0 Dueling Shield Thrusting Shields Standard +8 183.0 183.0 9.0 Dueling Shield Thrusting Shields Standard +9 190.0 190.0 9.0 Dueling Shield Thrusting Shields Standard +10 197.0 197.0 9.0 Dueling Shield Thrusting Shields Standard +11 204.0 204.0 9.0 Dueling Shield Thrusting Shields Standard +12 212.0 212.0 9.0 Dueling Shield Thrusting Shields Standard +13 219.0 219.0 9.0 Dueling Shield Thrusting Shields Standard +14 226.0 226.0 9.0 Dueling Shield Thrusting Shields Standard +15 233.0 233.0 9.0 Dueling Shield Thrusting Shields Standard +16 241.0 241.0 9.0 Dueling Shield Thrusting Shields Standard +17 248.0 248.0 9.0 Dueling Shield Thrusting Shields Standard +18 255.0 255.0 9.0 Dueling Shield Thrusting Shields Standard +19 262.0 262.0 9.0

In [0]:
%sql
SELECT COUNT(DISTINCT weapon_name) AS total_weapon_stat_records FROM eldenringcatalog.gold.fact_weapon_stats;

total_weapon_stat_records 400

 **Understand the fact table structure with upgrade levels and detailed stats.**

# **11: What is the highest base physical attack damage?**

**Raw damage output at standard upgrade level.**


✔ All normal weapons start as STANDARD

✔ Affinities like Sacred, Flame, Cold, Fire, Keen, etc. modify the base form

✔ Unique/Somber weapons do not have affinities

✔ If upgrade level has no "+" and affinity is Standard → it’s the true base weapon

In [0]:
%sql
SELECT 
    fs.weapon_name,
    fs.category,
    w.is_dlc,
    fs.upgrade_level,
    fs.attack_physical,
    fs.weight,
    fs.total_attack_power
FROM eldenringcatalog.gold.fact_weapon_stats fs
JOIN eldenringcatalog.gold.dim_weapons w
ON fs.weapon_name = w.weapon_name
WHERE fs.upgrade_level LIKE '%Standard%'
AND fs.Upgrade_Level NOT LIKE '%+%'
ORDER BY fs.attack_physical DESC

weapon_name category is_dlc upgrade_level attack_physical weight total_attack_power Jar Cannon Ballistas 0 Standard 192.0 15.0 192.0 Duelist Greataxe Colossal Weapons 0 Standard 170.0 20.0 170.0 Axe of Godfrey Colossal Weapons 0 Standard 165.0 18.0 165.0 Rotten Staff Colossal Weapons 0 Standard 165.0 18.0 165.0 Watchdog's Staff Colossal Weapons 0 Standard 165.0 19.0 165.0 Greatsword Colossal Swords 0 Standard 164.0 23.0 164.0 Rotten Greataxe Colossal Weapons 0 Standard 162.0 20.0 162.0 Grafted Blade Greatsword Colossal Swords 0 Standard 162.0 21.0 162.0 Golem's Halberd Colossal Weapons 0 Standard 158.0 21.5 158.0 Ghiza's Wheel Colossal Weapons 0 Standard 156.0 19.0 156.0 Prelate's Inferno Crozier Colossal Weapons 0 Standard 156.0 23.5 156.0 Giant-Crusher Colossal Weapons 0 Standard 155.0 26.5 155.0 Troll's Golden Sword Colossal Swords 0 Standard 155.0 19.0 155.0 Rakshasa's Great Katana Great Katanas 1 Standard 155.0 9.5 155.0 Ancient Meteoric Ore Greatsword Colossal Swords 1 Standard 154.0 22.0 200.0 Great Club Colossal Weapons 0 Standard 154.0 17.0 200.0 Troll's Hammer Colossal Weapons 0 Standard 153.0 16.0 198.0 Dragon-Hunter's Great Katana Great Katanas 1 Standard 152.0 10.5 152.0 Greataxe Greataxes 0 Standard 151.0 13.0 151.0 Executioner's Greataxe Greataxes 0 Standard 150.0 15.0 150.0 Iron Greatsword Greatswords 0 Standard 149.0 12.0 149.0 Fire Knight's Greatsword Colossal Swords 1 Standard 147.0 16.0 191.0 Bloodfiend's Arm Colossal Weapons 1 Standard 147.0 20.0 147.0 Rusted Anchor Greataxes 0 Standard 147.0 12.5 147.0 Devonia's Hammer Colossal Weapons 1 Standard 147.0 20.0 191.0 Longhaft Axe Greataxes 0 Standard 146.0 12.5 146.0 Crescent Moon Axe Greataxes 0 Standard 146.0 12.5 146.0 Freyja's Greatsword Curved Greatswords 1 Standard 146.0 14.0 146.0 Curved Great Club Great Hammers 0 Standard 145.0 10.0 145.0 Great Katana Great Katanas 1 Standard 145.0 9.0 145.0 Grave Scythe Reapers 0 Standard 144.0 9.5 144.0 Pickaxe Great Hammers 0 Standard 144.0 10.0 144.0 Beastman's Cleaver Curved Greatswords 0 Standard 143.0 16.5 143.0 Watchdog's Greatsword Colossal Swords 0 Standard 142.0 22.0 142.0 Great Omenkiller Cleaver Greataxes 0 Standard 142.0 11.0 142.0 Axe of Godrick Greataxes 0 Standard 142.0 11.0 142.0 Banished Knight's Greatsword Greatswords 0 Standard 142.0 10.0 142.0 Omen Cleaver Curved Greatswords 0 Standard 142.0 11.5 142.0 Smithscript Greathammer Great Hammers 1 Standard 141.0 9.0 141.0 Bloodhound's Fang Curved Greatswords 0 Standard 141.0 11.5 141.0 Putrescence Cleaver Greataxes 1 Standard 141.0 13.5 183.0 Knight's Greatsword Greatswords 0 Standard 141.0 10.0 141.0 Zweihander Colossal Swords 0 Standard 141.0 15.5 141.0 Sword of Milos Greatswords 0 Standard 141.0 10.0 141.0 Lizard Greatsword Greatswords 1 Standard 140.0 9.5 140.0 Black Steel Greathammer Great Hammers 1 Standard 140.0 12.0 182.0 Guardian's Swordspear Halberds 0 Standard 139.0 9.0 139.0 Celebrant's Skull Great Hammers 0 Standard 138.0 8.5 138.0 Commander's Standard Halberds 0 Standard 138.0 11.5 138.0 Bastard Sword Greatswords 0 Standard 138.0 9.0 138.0 Dismounter Curved Greatswords 0 Standard 138.0 10.0 138.0 Claymore Greatswords 0 Standard 138.0 9.0 138.0 Stormhawk Axe Axes 0 Standard 136.0 5.5 136.0 Lordsworn's Greatsword Greatswords 0 Standard 136.0 9.0 136.0 Great Stars Great Hammers 0 Standard 135.0 10.0 135.0 Hand Ballista Ballistas 0 Standard 135.0 10.0 135.0 Rabbath's Cannon Ballistas 1 Standard 135.0 15.0 270.0 Great Stars Great Hammers 0 Standard 135.0 10.0 135.0 Dragon Halberd Halberds 0 Standard 135.0 10.5 135.0 Greathorn Hammer Great Hammers 0 Standard 134.0 10.0 134.0 Butchering Knife Greataxes 0 Standard 134.0 8.5 134.0 Bonny Butchering Knife Greataxes 1 Standard 134.0 8.5 134.0 Golden Halberd Halberds 0 Standard 134.0 13.5 221.0 Great Mace Great Hammers 0 Standard 134.0 11.5 134.0 Monk's Flameblade Curved Greatswords 0 Standard 134.0 9.0 134.0 Gazing Finger Colossal Weapons 1 Standard 133.0 15.0 219.0 Messmer Soldier's Axe Axes 1 Stand

**Colossal swords and other heavy weapons usually offer the highest physical damage, with the Jar Cannon having the highest base physical attack**

# **📊 12 - Calculate damage per weight ratio for all weapons**

**Efficiency metric for mobility-focused builds.**

In [0]:
%sql
SELECT 
    weapon_name,
    category,
    attack_physical,
    weight,
    ROUND(
        CASE 
            WHEN weight = 0 THEN NULL
            ELSE attack_physical / weight
        END, 2
    ) AS damage_per_weight
FROM eldenringcatalog.gold.fact_weapon_stats
WHERE upgrade_level LIKE '%Standard%'
AND upgrade_level NOT LIKE '%+%'
ORDER BY damage_per_weight DESC
LIMIT 30;

weapon_name category attack_physical weight damage_per_weight Thiollier's Hidden Needle Fists 70.0 0.5 140.0 Deadly Poison Perfume Bottle Perfume Bottles 92.0 1.0 92.0 Dane's Footwork Hand-to-Hand Arts 80.0 1.0 80.0 Dryleaf Arts Hand-to-Hand Arts 80.0 1.0 80.0 Poisoned Hand Fists 76.0 1.0 76.0 Caestus Fists 90.0 1.5 60.0 Raptor Talons Claws 86.0 1.5 57.33 Madding Hand Fists 53.0 1.0 53.0 Celebrant's Sickle Daggers 79.0 1.5 52.67 Backhand Blade Backhand Blades 105.0 2.0 52.5 Whip Whips 103.0 2.0 51.5 Great Knife Daggers 75.0 1.5 50.0 Smithscript Cirque Backhand Blades 100.0 2.0 50.0 Parrying Dagger Daggers 75.0 1.5 50.0 Dagger Daggers 74.0 1.5 49.33 Shotel Curved Swords 96.0 2.0 48.0 Blade of Calling Daggers 71.0 1.5 47.33 Hookclaws Claws 93.0 2.0 46.5 Spiked Caestus Fists 91.0 2.0 45.5 Forked Hatchet Axes 113.0 2.5 45.2 Warpick Hammers 108.0 2.5 43.2 Thorned Whip Whips 105.0 2.5 42.0 Smithscript Dagger Throwing Blades 62.0 1.5 41.33 Crystal Knife Daggers 82.0 2.0 41.0 Bloodstained Dagger Daggers 81.0 2.0 40.5 Iron Ball Fists 101.0 2.5 40.4 Frozen Needle Thrusting Swords 99.0 2.5 39.6 Katar Fists 97.0 2.5 38.8 Venomous Fang Claws 97.0 2.5 38.8 Rapier Thrusting Swords 96.0 2.5 38.4

In [0]:
%sql
SELECT 
    category,count( *) as count,
    MAX(
        CASE 
            WHEN weight = 0 THEN NULL
            ELSE ROUND(attack_physical / weight, 2)
        END
    ) AS best_efficiency
FROM eldenringcatalog.gold.fact_weapon_stats
WHERE upgrade_level LIKE '%Standard%'
AND upgrade_level NOT LIKE '%+%'
GROUP BY category
ORDER BY best_efficiency DESC;

category count best_efficiency Fists 14 140.0 Perfume Bottles 5 92.0 Hand-to-Hand Arts 2 80.0 Claws 5 57.33 Daggers 17 52.67 Backhand Blades 3 52.5 Whips 7 51.5 Curved Swords 19 48.0 Axes 17 45.2 Hammers 15 43.2 Throwing Blades 1 41.33 Thrusting Swords 8 39.6 Straight Swords 23 38.4 Torches 8 36.0 Spears 19 33.0 Beast Claws 2 32.67 Light Bows 5 31.0 Flails 6 22.4 Bows 9 22.0 Katanas 10 20.91 Sacred Seals 13 20.67 Halberds 18 18.62 Reapers 5 18.46 Light Greatswords 3 17.85 Twinblades 8 17.0 Great Katanas 3 16.32 Heavy Thrusting Swords 4 16.27 Crossbows 9 16.25 Great Hammers 18 16.24 Greatswords 25 16.0 Curved Greatswords 11 16.0 Greataxes 14 15.76 Great Spears 10 14.38 Thrusting Shields 2 13.89 Ballistas 3 13.5 Greatbows 4 12.63 Glintstone Staffs 17 11.6 Glinstone Staves 2 10.8 Colossal Weapons 20 9.56 Colossal Swords 17 9.19 null 4 null

**Light weapons often have better damage-per-weight ratios, favoring agile builds**

# **📊 13: Which weapons have magical damage (magic, fire, lightning, holy)?**

**Elemental damage for different enemy resistances.**

In [0]:
%sql
SELECT 
    weapon_name,
    category,
    attack_physical,
    attack_magic,
    attack_fire,
    attack_lightning,
    attack_holy,
    total_attack_power
FROM eldenringcatalog.gold.fact_weapon_stats
WHERE upgrade_level LIKE '%Standard%'
AND upgrade_level NOT LIKE '%+%'

  AND (attack_magic > 0 OR attack_fire > 0 OR attack_lightning > 0 OR attack_holy > 0)
ORDER BY total_attack_power DESC
LIMIT 30;

weapon_name category attack_physical attack_magic attack_fire attack_lightning attack_holy total_attack_power Rabbath's Cannon Ballistas 135.0 135.0 0.0 0.0 0.0 270.0 Royal Greatsword Colossal Swords 112.0 134.0 0.0 0.0 0.0 246.0 Rellana's Twin Blades Light Greatswords 103.0 66.0 66.0 0.0 0.0 235.0 Shadow Sunflower Blossom Colossal Weapons 114.0 0.0 0.0 0.0 114.0 228.0 Gargoyle's Black Halberd Halberds 103.0 0.0 0.0 0.0 123.0 226.0 Gargoyle's Blackblade Greatswords 102.0 0.0 0.0 0.0 122.0 224.0 Golden Halberd Halberds 134.0 0.0 0.0 0.0 87.0 221.0 Gargoyle's Black Axe Greataxes 100.0 0.0 0.0 0.0 120.0 220.0 Gazing Finger Colossal Weapons 133.0 86.0 0.0 0.0 0.0 219.0 Fallingstar Beast Jaw Colossal Weapons 131.0 85.0 0.0 0.0 0.0 216.0 Anvil Hammer Colossal Weapons 131.0 0.0 85.0 0.0 0.0 216.0 Starscourge Greatsword Colossal Swords 129.0 83.0 0.0 0.0 0.0 212.0 Maliketh's Black Blade Colossal Swords 127.0 0.0 0.0 0.0 82.0 209.0 Greatsword of Radahn (Light) Colossal Swords 127.0 82.0 0.0 0.0 0.0 209.0 Greatsword of Radahn (Lord) Colossal Swords 127.0 82.0 0.0 0.0 0.0 209.0 Maliketh's Black Blade Colossal Swords 127.0 0.0 0.0 0.0 82.0 209.0 Troll Knight's Sword Colossal Swords 124.0 80.0 0.0 0.0 0.0 204.0 Moonrithyll's Knight Sword Colossal Swords 124.0 80.0 0.0 0.0 0.0 204.0 Helphen's Steeple Greatswords 101.0 101.0 0.0 0.0 0.0 202.0 Greatsword of Damnation Greatswords 123.0 0.0 0.0 0.0 79.0 202.0 Treespear Great Spears 122.0 0.0 0.0 0.0 79.0 201.0 Great Club Colossal Weapons 154.0 0.0 0.0 0.0 46.0 200.0 Ancient Meteoric Ore Greatsword Colossal Swords 154.0 46.0 0.0 0.0 0.0 200.0 Sword of Night and Flame Straight Swords 87.0 56.0 56.0 0.0 0.0 199.0 Envoy's Greathorn Colossal Weapons 121.0 0.0 0.0 0.0 78.0 199.0 Blasphemous Blade Greatswords 121.0 0.0 78.0 0.0 0.0 199.0 Dragon Greatclaw Colossal Weapons 120.0 0.0 0.0 78.0 0.0 198.0 Troll's Hammer Colossal Weapons 153.0 0.0 45.0 0.0 0.0 198.0 Envoy's Long Horn Great Hammers 120.0 0.0 0.0 0.0 78.0 198.0 Cranial Vessel Candlestand Great Hammers 98.0 0.0 98.0 0.0 0.0 196.0

In [0]:
%sql
SELECT 
    SUM(CASE WHEN attack_magic > 0 THEN 1 ELSE 0 END) AS magic_weapons,
    SUM(CASE WHEN attack_fire > 0 THEN 1 ELSE 0 END) AS fire_weapons,
    SUM(CASE WHEN attack_lightning > 0 THEN 1 ELSE 0 END) AS lightning_weapons,
    SUM(CASE WHEN attack_holy > 0 THEN 1 ELSE 0 END) AS holy_weapons
FROM eldenringcatalog.gold.fact_weapon_stats
WHERE upgrade_level LIKE '%Standard%'
AND upgrade_level NOT LIKE '%+%'

magic_weapons fire_weapons lightning_weapons holy_weapons 55 32 7 46

**Specialized weapons split damage between physical and elemental for tactical advantages.**

# **📊  14 - Analyze scaling letters (S, A, B, C, D, E)**

**Understand how weapons benefit from stat investment.**

In [0]:
%sql
SELECT left(scaling_strength ,1) as scaling_strength, COUNT(*) as count
FROM eldenringcatalog.gold.fact_weapon_stats
WHERE upgrade_level LIKE '%Standard%'
AND upgrade_level NOT LIKE '%+%'
GROUP BY left(scaling_strength ,1)
ORDER BY scaling_strength DESC;

scaling_strength count E 112 D 209 C 50 B 6 - 28

In [0]:
%sql
SELECT left(scaling_dexterity,1) AS scaling_dexterity, COUNT(*) as count
FROM eldenringcatalog.gold.fact_weapon_stats
WHERE upgrade_level LIKE '%Standard%'
AND upgrade_level NOT LIKE '%+%'
GROUP BY left(scaling_dexterity,1)
ORDER BY scaling_dexterity;

scaling_dexterity count - 57 B 1 C 50 D 206 E 91

In [0]:
%sql
SELECT left(scaling_faith,1) AS scaling_faith, COUNT(*) as count
FROM eldenringcatalog.gold.fact_weapon_stats
WHERE upgrade_level LIKE '%Standard%'
AND upgrade_level NOT LIKE '%+%'
GROUP BY left(scaling_faith,1)
ORDER BY scaling_faith;

scaling_faith count - 307 B 1 C 17 D 64 E 16

In [0]:
%sql
SELECT left(scaling_intelligence,1) AS scaling_intelligence, COUNT(*) as count
FROM eldenringcatalog.gold.fact_weapon_stats
WHERE upgrade_level LIKE '%Standard%'
AND upgrade_level NOT LIKE '%+%'
GROUP BY left(scaling_intelligence,1)
ORDER BY scaling_intelligence;

scaling_intelligence count - 329 B 6 C 10 D 37 E 23

In [0]:
%sql
SELECT left(scaling_arcane,1) AS scaling_arcane, COUNT(*) as count
FROM eldenringcatalog.gold.fact_weapon_stats
WHERE upgrade_level LIKE '%Standard%'
AND upgrade_level NOT LIKE '%+%'
GROUP BY left(scaling_arcane,1)
ORDER BY scaling_arcane;

scaling_arcane count - 381 A 1 B 1 C 3 D 19

**distribution of scaling grades and identify which stats have best scaling options.**

# **📊  15: Which weapons have S-tier scaling in any stat?**

In [0]:
%sql
SELECT 
    weapon_name,
    category,
    upgrade_level,
    scaling_strength,
    scaling_dexterity,
    scaling_intelligence,
    scaling_faith,
    scaling_arcane,
    attack_physical
FROM eldenringcatalog.gold.fact_weapon_stats
WHERE upgrade_level LIKE '%Standard%'
  AND (scaling_strength = 'S' OR scaling_dexterity = 'S' OR 
       scaling_intelligence = 'S' OR scaling_faith = 'S' OR scaling_arcane = 'S')
ORDER BY weapon_name;

weapon_name category upgrade_level scaling_strength scaling_dexterity scaling_intelligence scaling_faith scaling_arcane attack_physical Academy Glintstone Staff Glintstone Staffs Standard +19 D - S - - 39.0 Academy Glintstone Staff Glintstone Staffs Standard +13 E - S - - 34.0 Academy Glintstone Staff Glintstone Staffs Standard +23 D - S - - 42.0 Academy Glintstone Staff Glintstone Staffs Standard +18 D - S - - 38.0 Academy Glintstone Staff Glintstone Staffs Standard +16 D - S - - 37.0 Academy Glintstone Staff Glintstone Staffs Standard +20 D - S - - 40.0 Academy Glintstone Staff Glintstone Staffs Standard +17 D - S - - 37.0 Academy Glintstone Staff Glintstone Staffs Standard +25 D - S - - 43.0 Academy Glintstone Staff Glintstone Staffs Standard +14 D - S - - 35.0 Academy Glintstone Staff Glintstone Staffs Standard +15 D - S - - 36.0 Academy Glintstone Staff Glintstone Staffs Standard +24 D - S - - 43.0 Academy Glintstone Staff Glintstone Staffs Standard +21 D - S - - 40.0 Academy Glintstone Staff Glintstone Staffs Standard +12 E - S - - 34.0 Academy Glintstone Staff Glintstone Staffs Standard +22 D - S - - 41.0 Astrologer's Staff Glintstone Staffs Standard +20 D - S - - 40.0 Astrologer's Staff Glintstone Staffs Standard +23 D - S - - 42.0 Astrologer's Staff Glintstone Staffs Standard +17 D - S - - 37.0 Astrologer's Staff Glintstone Staffs Standard +21 D - S - - 40.0 Astrologer's Staff Glintstone Staffs Standard +13 E - S - - 34.0 Astrologer's Staff Glintstone Staffs Standard +22 D - S - - 41.0 Astrologer's Staff Glintstone Staffs Standard +24 D - S - - 43.0 Astrologer's Staff Glintstone Staffs Standard +15 D - S - - 36.0 Astrologer's Staff Glintstone Staffs Standard +18 D - S - - 38.0 Astrologer's Staff Glintstone Staffs Standard +14 D - S - - 35.0 Astrologer's Staff Glintstone Staffs Standard +19 D - S - - 39.0 Astrologer's Staff Glintstone Staffs Standard +25 D - S - - 43.0 Astrologer's Staff Glintstone Staffs Standard +16 D - S - - 37.0 Azur's Glintstone Staff Glintstone Staffs Standard +10 C - S - - 58.0 Azur's Glintstone Staff Glintstone Staffs Standard +7 C - S - - 48.0 Azur's Glintstone Staff Glintstone Staffs Standard +8 C - S - - 51.0 Azur's Glintstone Staff Glintstone Staffs Standard +6 D - S - - 44.0 Azur's Glintstone Staff Glintstone Staffs Standard +5 D - S - - 41.0 Azur's Glintstone Staff Glintstone Staffs Standard +9 C - S - - 55.0 Carian Glintblade Staff Glintstone Staffs Standard +23 D D S - - 37.0 Carian Glintblade Staff Glintstone Staffs Standard +15 D E S - - 31.0 Carian Glintblade Staff Glintstone Staffs Standard +17 D E S - - 33.0 Carian Glintblade Staff Glintstone Staffs Standard +25 D D S - - 38.0 Carian Glintblade Staff Glintstone Staffs Standard +14 D E S - - 31.0 Carian Glintblade Staff Glintstone Staffs Standard +16 D E S - - 32.0 Carian Glintblade Staff Glintstone Staffs Standard +24 D D S - - 37.0 Carian Glintblade Staff Glintstone Staffs Standard +20 D D S - - 35.0 Carian Glintblade Staff Glintstone Staffs Standard +19 D D S - - 34.0 Carian Glintblade Staff Glintstone Staffs Standard +13 E E S - - 30.0 Carian Glintblade Staff Glintstone Staffs Standard +22 D D S - - 36.0 Carian Glintblade Staff Glintstone Staffs Standard +18 D E S - - 33.0 Carian Glintblade Staff Glintstone Staffs Standard +21 D D S - - 35.0 Carian Glintstone Staff Glintstone Staffs Standard +21 D - S - - 40.0 Carian Glintstone Staff Glintstone Staffs Standard +20 D - S - - 40.0 Carian Glintstone Staff Glintstone Staffs Standard +22 D - S - - 41.0 Carian Glintstone Staff Glintstone Staffs Standard +14 D - S - - 35.0 Carian Glintstone Staff Glintstone Staffs Standard +15 D - S - - 36.0 Carian Glintstone Staff Glintstone Staffs Standard +24 D - S - - 43.0 Carian Glintstone Staff Glintstone Staffs Standard +17 D - S - - 37.0 Carian Glintstone Staff Glintstone Staffs Standard +19 D - S - - 39.0 Carian Glintstone Staff Glintstone Staffs Standard +16 D - S - - 37.0 Carian Glintstone Staff Glintstone Staffs Standard +23 D - S - - 42.0 

**Identify premium weapons that maximize returns on stat investment.**

# **📊 16 - What is the average stamina cost per weapon category?**

**Stamina management varies by weapon type.**

In [0]:
%sql
SELECT 
    category,
    ROUND(AVG(stamina_cost), 2) AS avg_stamina,
    MIN(stamina_cost) AS min_stamina,
    MAX(stamina_cost) AS max_stamina,
    COUNT(*) AS weapon_count
FROM eldenringcatalog.gold.fact_weapon_stats
WHERE upgrade_level LIKE '%Standard%'
AND upgrade_level NOT LIKE '%+'
GROUP BY category
ORDER BY avg_stamina DESC;

category avg_stamina min_stamina max_stamina weapon_count Great Hammers 142.22 77.0 218.0 408 Colossal Weapons 140.49 72.0 224.0 340 Greataxes 124.8 80.0 184.0 274 Ballistas 115.94 70.0 180.0 48 Colossal Swords 112.71 60.0 168.0 262 Greatbows 112.37 75.0 150.0 59 Thrusting Shields 108.56 70.0 150.0 52 Curved Greatswords 104.14 48.0 704.0 196 Greatswords 100.13 59.0 168.0 425 Halberds 96.65 57.0 152.0 363 Great Spears 89.65 40.0 146.0 155 Axes 86.21 47.0 128.0 382 Flails 85.42 53.0 122.0 111 Reapers 84.24 52.0 120.0 85 Hammers 81.67 42.0 130.0 285 Light Greatswords 81.48 53.0 114.0 48 Great Katanas 78.65 52.0 110.0 48 Fists 77.27 45.0 118.0 259 Hand-to-Hand Arts 75.0 50.0 100.0 52 Torches 74.42 42.0 120.0 148 Heavy Thrusting Swords 74.22 46.0 110.0 74 Backhand Blades 70.52 45.0 116.0 63 Twinblades 69.37 42.0 100.0 163 Curved Swords 68.69 40.0 102.0 374 Straight Swords 67.47 38.0 122.0 373 null 67.04 28.0 116.0 104 Katanas 66.61 42.0 104.0 155 Whips 65.8 42.0 92.0 137 Perfume Bottles 62.55 42.0 84.0 115 Spears 59.54 34.0 88.0 404 Beast Claws 54.89 36.0 78.0 37 Claws 54.62 34.0 78.0 115 Crossbows 46.76 28.0 68.0 174 Throwing Blades 44.62 30.0 60.0 26 Daggers 44.47 28.0 68.0 352 Bows 43.65 20.0 60.0 144 Thrusting Swords 40.2 0.0 66.0 193 Light Bows 29.65 20.0 40.0 115 Glinstone Staves 22.86 0.0 44.0 37 Sacred Seals 8.78 0.0 40.0 263 Glintstone Staffs 3.12 0.0 46.0 367

**Heavier weapon categories require more stamina per swing, affecting combat rhythm.**

# **17 - Analyze guard stats (physical block percentage)**

**Some weapons can block attacks effectively.**

In [0]:
%sql
SELECT 
    weapon_name,
    category,
    guard_physical,
    guard_boost,
    guard_resistance
FROM eldenringcatalog.gold.fact_weapon_stats
WHERE upgrade_level LIKE '%Standard%' AND upgrade_level NOT LIKE '%+' AND guard_physical > 0
ORDER BY guard_physical DESC
LIMIT 30;

weapon_name category guard_physical guard_boost guard_resistance Carian Thrusting Shield Thrusting Shields 94.0 74.24 21.0 Carian Thrusting Shield Thrusting Shields 94.0 64.64 21.0 Carian Thrusting Shield Thrusting Shields 94.0 69.76 21.0 Carian Thrusting Shield Thrusting Shields 94.0 74.24 21.0 Carian Thrusting Shield Thrusting Shields 94.0 76.16 21.0 Carian Thrusting Shield Thrusting Shields 94.0 65.28 21.0 Carian Thrusting Shield Thrusting Shields 94.0 67.2 21.0 Carian Thrusting Shield Thrusting Shields 94.0 69.12 21.0 Carian Thrusting Shield Thrusting Shields 94.0 71.68 21.0 Carian Thrusting Shield Thrusting Shields 94.0 73.6 21.0 Carian Thrusting Shield Thrusting Shields 94.0 75.52 21.0 Carian Thrusting Shield Thrusting Shields 94.0 75.52 21.0 Carian Thrusting Shield Thrusting Shields 94.0 76.16 21.0 Carian Thrusting Shield Thrusting Shields 94.0 76.8 21.0 Carian Thrusting Shield Thrusting Shields 94.0 65.92 21.0 Carian Thrusting Shield Thrusting Shields 94.0 65.92 21.0 Carian Thrusting Shield Thrusting Shields 94.0 67.84 21.0 Carian Thrusting Shield Thrusting Shields 94.0 68.48 21.0 Carian Thrusting Shield Thrusting Shields 94.0 70.4 21.0 Carian Thrusting Shield Thrusting Shields 94.0 71.04 21.0 Carian Thrusting Shield Thrusting Shields 94.0 72.32 21.0 Carian Thrusting Shield Thrusting Shields 94.0 72.96 21.0 Carian Thrusting Shield Thrusting Shields 94.0 74.88 21.0 Carian Thrusting Shield Thrusting Shields 94.0 74.88 21.0 Carian Thrusting Shield Thrusting Shields 94.0 75.52 21.0 Carian Thrusting Shield Thrusting Shields 94.0 64.0 21.0 Greatsword of Solitude Greatswords 89.0 53.0 25.0 Greatsword of Solitude Greatswords 89.0 53.53 25.0 Greatsword of Solitude Greatswords 89.0 54.06 25.0 Greatsword of Solitude Greatswords 89.0 54.59 25.0

In [0]:
%sql
SELECT 
    category,
    COUNT(*) AS guardable_weapons,
    ROUND(AVG(guard_physical), 2) AS avg_guard_physical
FROM eldenringcatalog.gold.fact_weapon_stats
WHERE upgrade_level LIKE '%Standard%' AND guard_physical > 0
GROUP BY category
ORDER BY avg_guard_physical DESC;

category guardable_weapons avg_guard_physical Thrusting Shields 52 91.0 Colossal Weapons 340 74.41 Colossal Swords 262 72.9 Curved Greatswords 196 67.62 Great Hammers 408 67.49 Greatswords 425 65.6 Greataxes 274 64.53 Light Greatswords 48 63.69 Great Katanas 48 60.46 Heavy Thrusting Swords 74 58.53 Halberds 363 57.33 Great Spears 155 53.87 null 104 48.5 Axes 382 47.09 Thrusting Swords 193 46.6 Spears 404 45.9 Hammers 285 44.55 Reapers 85 43.51 Twinblades 163 43.07 Katanas 155 43.05 Backhand Blades 63 42.62 Curved Swords 374 41.78 Straight Swords 373 40.42 Flails 111 37.38 Fists 259 36.47 Beast Claws 37 35.89 Claws 115 35.26 Hand-to-Hand Arts 52 35.0 Throwing Blades 26 35.0 Daggers 352 34.43 Whips 137 25.9 Glintstone Staffs 367 25.04 Sacred Seals 263 25.0 Torches 148 23.91 Glinstone Staves 37 23.19 Perfume Bottles 115 22.09

**Some weapon categories offer viable blocking, though shields are superior.**

# **📊 18 - Compare base game vs DLC weapon damage**

**DLC power creep analysis.**

In [0]:
%sql
SELECT 
    CASE WHEN w.is_dlc = 1 THEN 'DLC' ELSE 'Base Game' END AS content_type,
    ROUND(AVG(ws.attack_physical), 2) AS avg_physical_dmg,
    ROUND(AVG(ws.total_attack_power), 2) AS avg_total_dmg,
    ROUND(AVG(ws.weight), 2) AS avg_weight,
    COUNT(*) AS weapon_count
FROM eldenringcatalog.gold.fact_weapon_stats ws
INNER JOIN eldenringcatalog.gold.dim_weapons w ON ws.weapon_name = w.weapon_name
WHERE ws.upgrade_level LIKE '%Standard%'
AND ws.upgrade_level NOT LIKE '%+%'
GROUP BY w.is_dlc;

content_type avg_physical_dmg avg_total_dmg avg_weight weapon_count DLC 98.91 131.35 6.94 93 Base Game 101.51 124.28 7.25 308

**Determine if DLC weapons have higher average stats (power creep).**

# **📊 19 - Which weapon has the highest total attack power at max upgrade?**

In [0]:
%sql
SELECT weapon_name,
       category,
       upgrade_level,
       total_attack_power,
       attack_physical,
       attack_magic,
       attack_fire,
       attack_lightning,
       attack_holy
FROM (
    SELECT *,
           ROW_NUMBER() OVER (PARTITION BY weapon_name ORDER BY total_attack_power DESC) AS rn
    FROM eldenringcatalog.gold.fact_weapon_stats
) t
WHERE rn = 1
ORDER BY total_attack_power DESC
LIMIT 20;


weapon_name category upgrade_level total_attack_power attack_physical attack_magic attack_fire attack_lightning attack_holy Carian Sorcery Sword Thrusting Swords Standard +18 1348.0 1224.0 124.0 0.0 0.0 0.0 Rabbath's Cannon Ballistas Standard +10 944.0 472.0 472.0 0.0 0.0 0.0 Jar Cannon Ballistas Standard +10 672.0 672.0 0.0 0.0 0.0 0.0 Duelist Greataxe Colossal Weapons Flame +25 656.0 328.0 0.0 328.0 0.0 0.0 Greatsword Colossal Swords Flame +25 636.0 318.0 0.0 318.0 0.0 0.0 Rotten Greataxe Colossal Weapons Flame +25 628.0 314.0 0.0 314.0 0.0 0.0 Black Steel Greathammer Great Hammers Flame +25 616.0 266.0 0.0 266.0 0.0 84.0 Golem's Halberd Colossal Weapons Flame +25 612.0 306.0 0.0 306.0 0.0 0.0 Prelate's Inferno Crozier Colossal Weapons Flame +25 604.0 302.0 0.0 302.0 0.0 0.0 Royal Greatsword Colossal Swords Standard +10 602.0 274.0 328.0 0.0 0.0 0.0 Troll's Golden Sword Colossal Swords Flame +25 600.0 300.0 0.0 300.0 0.0 0.0 Hand Ballista Ballistas Standard +25 600.0 600.0 0.0 0.0 0.0 0.0 Giant-Crusher Colossal Weapons Flame +25 600.0 300.0 0.0 300.0 0.0 0.0 Greataxe Greataxes Flame +25 584.0 292.0 0.0 292.0 0.0 0.0 Executioner's Greataxe Greataxes Flame +25 580.0 290.0 0.0 290.0 0.0 0.0 Fire Knight's Greatsword Colossal Swords Sacred +25 578.0 254.0 0.0 70.0 0.0 254.0 Iron Greatsword Greatswords Flame +25 576.0 288.0 0.0 288.0 0.0 0.0 Rellana's Twin Blades Light Greatswords Standard +10 574.0 252.0 161.0 161.0 0.0 0.0 Bloodfiend's Arm Colossal Weapons Flame +25 568.0 284.0 0.0 284.0 0.0 0.0 Rusted Anchor Greataxes Flame +25 568.0 284.0 0.0 284.0 0.0 0.0

**Find the absolute highest damage weapons at maximum upgrade**

# **📊 21: Load status effects table and analyze bleed buildup**

**Hemorrhage (bleed) is one of the strongest status effects.**

In [0]:
%sql
SELECT 
    weapon_name,
    buildup_primary,
    buildup_secondary,
    upgrade_level,
    effect_type
FROM eldenringcatalog.gold.fact_status_effects
WHERE effect_type = 'bleed' AND upgrade_level like '%Standard%'
AND upgrade_level not like '%+%'
ORDER BY buildup_primary DESC

weapon_name buildup_primary buildup_secondary upgrade_level effect_type Ghiza's Wheel 70 0 Standard bleed Varre's Bouquet 65 0 Standard bleed Raptor Talons 60 0 Standard bleed Morgott's Cursed Sword 60 0 Standard bleed Bloodhound Claws 60 0 Standard bleed Hookclaws 60 0 Standard bleed Falx 55 0 Standard bleed Obsidian Lamina 55 0 Standard bleed Sword of Milos 55 0 Standard bleed Great Stars 55 0 Standard bleed Grave Scythe 55 0 Standard bleed Rakshasa's Great Katana 55 0 Standard bleed Flamberge 55 0 Standard bleed Halo Scythe 55 0 Standard bleed Forked Greatsword 55 0 Standard bleed Dragon-Hunter's Great Katana 55 0 Standard bleed Great Omenkiller Cleaver 55 0 Standard bleed Great Stars 55 0 Standard bleed Barbed Staff-Spear 55 0 Standard bleed Bloody Helice 55 0 Standard bleed Bloodfiend's Sacred Spear 55 0 Standard bleed Great Katana 55 0 Standard bleed Bloodhound's Fang 55 0 Standard bleed Winged Scythe 55 0 Standard bleed Eleonora's Poleblade 55 0 Standard bleed Hoslow's Petal Whip 55 0 Standard bleed Vulgar Militia Saw 55 0 Standard bleed Scythe 55 0 Standard bleed Mohgwyn's Sacred Spear 55 0 Standard bleed Spiked Club 50 0 Standard bleed Forked Hatchet 50 0 Standard bleed Forked-Tongue Hatchet 50 0 Standard bleed Thorned Whip 50 0 Standard bleed Hand of Malenia 50 0 Standard bleed Moonveil 50 0 Standard bleed Rivers of Blood 50 0 Standard bleed Inquisitor's Girandole 50 0 Standard bleed Meteoric Ore Blade 50 0 Standard bleed Spiked Spear 50 0 Standard bleed Bloodfiend's Fork 50 0 Standard bleed Reduvia 50 0 Standard bleed Cross-Naginata 50 0 Standard bleed Chainlink Flail 50 0 Standard bleed Morning Star 50 0 Standard bleed Flail 50 0 Standard bleed Nightrider Flail 50 0 Standard bleed Red Bear's Claw 45 0 Standard bleed Spiked Caestus 45 0 Standard bleed Beast Claw (Weapon) 45 0 Standard bleed Star Fist 45 0 Standard bleed Scavenger's Curved Sword 45 0 Standard bleed Star-Lined Sword 45 0 Standard bleed Sword of Night 45 0 Standard bleed Claws of Night 45 0 Standard bleed Nagakiba 45 0 Standard bleed Curseblade's Cirque 45 0 Standard bleed Uchigatana 45 0 Standard bleed Great Knife 38 0 Standard bleed Bloodstained Dagger 38 0 Standard bleed Wakizashi 38 0 Standard bleed

In [0]:
%sql
SELECT AVG(buildup_primary) AS avg_bleed_buildup
FROM eldenringcatalog.gold.fact_status_effects
WHERE effect_type = 'bleed' AND upgrade_level LIKE '%Standard%'
AND upgrade_level not like '%+%'

avg_bleed_buildup 51.65

**Identify fastest-bleeding weapons for hemorrhage builds.**

# **📊 22: Which weapons apply frostbite?**

In [0]:
%sql
SELECT 
    weapon_name,
    buildup_primary,
    buildup_secondary,
    upgrade_level,
    effect_type
FROM eldenringcatalog.gold.fact_status_effects
WHERE effect_type = 'frostbite' AND upgrade_level like '%Standard%'
AND upgrade_level not like '%+%'
ORDER BY buildup_primary DESC

weapon_name buildup_primary buildup_secondary upgrade_level effect_type Death's Poker 65 0 Standard frostbite Icerind Hatchet 65 0 Standard frostbite Ghostflame Torch 65 0 Standard frostbite Zamor Curved Sword 65 0 Standard frostbite Chilling Perfume Bottle 60 0 Standard frostbite Frozen Needle 60 0 Standard frostbite Dark Moon Greatsword 55 0 Standard frostbite Putrescence Cleaver 55 0 Standard frostbite

In [0]:
%sql
SELECT AVG(buildup_primary) AS avg_bleed_buildup
FROM eldenringcatalog.gold.fact_status_effects
WHERE effect_type = 'frostbite' AND upgrade_level LIKE '%Standard%'
AND upgrade_level not like '%+%'

avg_bleed_buildup 61.25

# **📊 23: Analyze poison buildup weapons**

In [0]:
%sql
SELECT 
    weapon_name,
    buildup_primary,
    buildup_secondary,
    upgrade_level,
    effect_type
FROM eldenringcatalog.gold.fact_status_effects
WHERE effect_type = 'poison' AND upgrade_level like '%Standard%'
AND upgrade_level not like '%+%'
ORDER BY buildup_primary DESC

weapon_name buildup_primary buildup_secondary upgrade_level effect_type Venomous Fang 72 0 Standard poison Serpentbone Blade 66 0 Standard poison Deadly Poison Perfume Bottle 66 0 Standard poison Tooth Whip 66 0 Standard poison Poisoned Hand 60 0 Standard poison

In [0]:
%sql
SELECT AVG(buildup_primary) AS avg_bleed_buildup
FROM eldenringcatalog.gold.fact_status_effects
WHERE effect_type = 'frostbite' AND upgrade_level LIKE '%Standard%'
AND upgrade_level not like '%+%'

avg_bleed_buildup 61.25

# **📊 24: Which weapons have dual status effects?**

**Some weapons apply multiple status effects simultaneously.**

In [0]:
%sql
SELECT 
    weapon_name,
    buildup_primary,
    buildup_secondary,
    upgrade_level,
    effect_type
FROM eldenringcatalog.gold.fact_status_effects
WHERE upgrade_level not like '%+%' AND has_dual_buildup = true
ORDER BY buildup_primary DESC

weapon_name buildup_primary buildup_secondary upgrade_level effect_type Grave Scythe 80 50 Poison poison Great Stars 80 50 Cold frostbite Great Stars 80 50 Poison poison Scythe 80 50 Cold frostbite Great Katana 80 50 Poison poison Great Stars 80 50 Poison poison Great Omenkiller Cleaver 80 50 Cold frostbite Great Stars 80 50 Cold frostbite Great Katana 80 50 Cold frostbite Flamberge 80 50 Poison poison Vulgar Militia Saw 80 50 Poison poison Scythe 80 50 Poison poison Great Omenkiller Cleaver 80 50 Poison poison Forked Greatsword 80 50 Poison poison Grave Scythe 80 50 Cold frostbite Vulgar Militia Saw 80 50 Cold frostbite Flamberge 80 50 Cold frostbite Forked Greatsword 80 50 Cold frostbite Thorned Whip 66 45 Poison poison Forked Hatchet 66 45 Poison poison Hoslow's Petal Whip 66 50 Cold frostbite Cross-Naginata 66 45 Poison poison Spiked Spear 66 45 Cold frostbite Forked Hatchet 66 45 Cold frostbite Cross-Naginata 66 45 Cold frostbite Spiked Spear 66 45 Poison poison Bloodfiend's Fork 66 45 Cold frostbite Great Knife 66 38 Poison poison Morning Star 66 45 Cold frostbite Thorned Whip 66 45 Cold frostbite Hoslow's Petal Whip 66 50 Poison poison Hookclaws 66 45 Cold frostbite Nagakiba 66 38 Cold frostbite Flail 66 45 Poison poison Bloodstained Dagger 66 38 Poison poison Beast Claw (Weapon) 66 38 Cold frostbite Uchigatana 66 38 Cold frostbite Nagakiba 66 38 Poison poison Nightrider Flail 66 45 Cold frostbite Chainlink Flail 66 45 Poison poison Scavenger's Curved Sword 66 38 Cold frostbite Venomous Fang 66 57 Blood poison Raptor Talons 66 45 Cold frostbite Venomous Fang 66 66 Cold frostbite Uchigatana 66 38 Poison poison Beast Claw (Weapon) 66 38 Poison poison Raptor Talons 66 45 Poison poison Wakizashi 66 38 Poison poison Scavenger's Curved Sword 66 38 Poison poison Morning Star 66 45 Poison poison Hookclaws 66 45 Poison poison Star Fist 66 38 Poison poison Bloodhound Claws 66 45 Cold frostbite Spiked Caestus 66 38 Poison poison Nightrider Flail 66 45 Poison poison Star Fist 66 38 Cold frostbite Bloodhound Claws 66 45 Poison poison Chainlink Flail 66 45 Cold frostbite Great Knife 66 38 Cold frostbite Spiked Club 66 45 Poison poison Spiked Club 66 45 Cold frostbite Bloodfiend's Fork 66 45 Poison poison Bloodstained Dagger 66 38 Cold frostbite Flail 66 45 Cold frostbite Wakizashi 66 38 Cold frostbite Spiked Caestus 66 38 Cold frostbite Rotten Battle Hammer 65 69 Blood scarlet_rot Rotten Greataxe 65 96 Cold scarlet_rot Rotten Greataxe 65 96 Poison scarlet_rot Rotten Greataxe 65 83 Blood scarlet_rot Rotten Battle Hammer 65 80 Cold scarlet_rot Rotten Battle Hammer 65 80 Poison scarlet_rot Antspur Rapier 50 66 Cold scarlet_rot Antspur Rapier 50 66 Poison scarlet_rot Antspur Rapier 50 57 Blood scarlet_rot

**Rare weapons that apply two status effects are highly valuable.**

# **📊 Question 25: Calculate average upgrade progression for physical damage**

**Understand damage scaling through upgrade levels.**

In [0]:
%sql
-- 1) Extract numeric upgrade number for Standard rows
WITH base AS (
    SELECT
        weapon_name,
        upgrade_level,
        TRY_CAST(regexp_extract(upgrade_level, '\\+(\\d+)', 1) AS INT) AS upgrade_number,
        attack_physical
    FROM eldenringcatalog.gold.fact_weapon_stats
    WHERE upgrade_level LIKE 'Standard%'
      AND regexp_extract(upgrade_level, '\\+(\\d+)', 1) <> ''   -- only rows containing +N
),

-- 2) Keep rows with valid upgrade_number
clean AS (
    SELECT *
    FROM base
    WHERE upgrade_number IS NOT NULL
),

-- 3) Determine the max Standard upgrade number per weapon
weapon_max AS (
    SELECT
        weapon_name,
        MAX(upgrade_number) AS max_upgrade
    FROM clean
    GROUP BY weapon_name
),

-- 4) Assign each weapon to its upgrade path (Standard+10 or Standard+25)
with_path AS (
    SELECT
        c.weapon_name,
        c.upgrade_level,
        c.upgrade_number,
        c.attack_physical,
        wm.max_upgrade,
        CONCAT('Standard+', CAST(wm.max_upgrade AS STRING)) AS upgrade_path
    FROM clean c
    JOIN weapon_max wm
      ON c.weapon_name = wm.weapon_name
),

-- 5) Aggregate: average physical damage per upgrade_number per path
agg AS (
    SELECT
        upgrade_path,
        upgrade_number,
        ROUND(AVG(attack_physical), 2) AS avg_physical_damage,
        COUNT(*) AS weapon_count
    FROM with_path
    GROUP BY upgrade_path, upgrade_number
),

-- 6) Compute percent progression within each (Standard+X) path
progression AS (
    SELECT
        upgrade_path,
        upgrade_number,
        avg_physical_damage,
        LAG(avg_physical_damage) OVER (
            PARTITION BY upgrade_path ORDER BY upgrade_number
        ) AS prev_damage
    FROM agg
)

-- 7) Final output
SELECT
    upgrade_path,
    upgrade_number,
    avg_physical_damage,
    ROUND(
        CASE
            WHEN prev_damage IS NULL THEN 0
            ELSE ((avg_physical_damage - prev_damage) / prev_damage) * 100
        END,
        2
    ) AS pct_increase
FROM progression
ORDER BY upgrade_path, upgrade_number;


upgrade_path upgrade_number avg_physical_damage pct_increase Standard+10 1 112.11 0.0 Standard+10 2 126.58 12.91 Standard+10 3 141.09 11.46 Standard+10 4 155.55 10.25 Standard+10 5 170.1 9.35 Standard+10 6 184.54 8.49 Standard+10 7 199.02 7.85 Standard+10 8 213.52 7.29 Standard+10 9 228.0 6.78 Standard+10 10 242.52 6.37 Standard+25 1 109.06 0.0 Standard+25 2 115.18 5.61 Standard+25 3 121.26 5.28 Standard+25 4 127.38 5.05 Standard+25 5 133.4 4.73 Standard+25 6 139.52 4.59 Standard+25 7 145.61 4.36 Standard+25 8 151.74 4.21 Standard+25 9 157.77 3.97 Standard+25 10 163.85 3.85 Standard+25 11 169.97 3.74 Standard+25 12 176.08 3.59 Standard+25 13 182.12 3.43 Standard+25 14 188.18 3.33 Standard+25 15 194.31 3.26 Standard+25 16 200.41 3.14 Standard+25 17 206.47 3.02 Standard+25 18 217.47 5.33 Standard+25 19 218.59 0.52 Standard+25 20 224.73 2.81 Standard+25 21 230.74 2.67 Standard+25 22 236.81 2.63 Standard+25 23 242.98 2.61 Standard+25 24 249.03 2.49 Standard+25 25 255.07 2.43

**See how damage scales with each upgrade level, plan upgrade priorities.**